packages/react-dom/src/client/ReactDOMSelection.js

```diff
@@ -151,9 +151,12 @@ export function getModernOffsetsFromPoints(
  * @param {object} offsets
  */
 export function setOffsets(node, offsets) {
-  const doc = node.ownerDocument || document;
-  const win = doc ? doc.defaultView : window;
-  const selection = win.getSelection();
+  const {ownerDocument} = node;
```

Comment:
Thanks for tracking this down! This is definitely an oversight of #12037 that we didn’t address.

For context, the reason that `getOffsets` uses a different method to get a relative `window` and `document` object than `setOffsets` is that `getOffsets` only needs to use the relative `window` (which has `getSelections`), whereas `setOffsets` needs to use both the `window` object (for `getSelection`, again), and the `document` object (for `createRange`). So this should really be a hybrid of the two approaches, i.e.:
```js
  const doc = node.ownerDocument || document;
  const win = (doc && doc.defaultView) || window;
  ...
```
Where it still guards against a missing `ownerDocument` by falling back to the global and also guards against the case  of having an `ownerDocument` that’s  missing a `defaultView`.

Reply:
I understand, I changed the logic a bit by over-protecting this path

Update incoming